presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -887,23 +887,37 @@ VeloxQueryPlanConverter::toVeloxQueryPlan(
   const auto type = toLocalExchangeType(node->type);
 
   const auto outputType = toRowType(node->partitioningScheme.outputLayout);
-  // Input layout from source, describing how data should be fed to our node.
-  // For all sources the layout should be the same, so we store only one (we
-  // use the 1st source for that).
-  // This layout and the output layout for the 1st source would be used to
-  // created the column mapping in the operator.
-  RowTypePtr inputTypeFromSource = toRowType(node->inputs[0]);
+
+  // Different source nodes may have different output layouts.
+  // Add ProjectNode on top of each source node to re-arrange the output columns
+  // to match the output layout of the LocalExchangeNode.
```

Comment:
I'm generally surprised that planner couldn't have that sorted out on the coordinator.

Reply:
I'm surprised as well. CC: @kaikalur